src/lib/mina.ts

```diff
@@ -1039,3 +1044,134 @@ async function verifyAccountUpdate(
     );
   }
 }
+
+function verifyTransactionLimits(accountUpdates: AccountUpdate[]) {
+  // constants used to calculate cost of a transaction - originally defined in the genesis_constants file in the mina repo
+  const proofCost = 10.26;
+  const signedPairCost = 10.08;
+  const signedSingleCost = 9.14;
+  const costLimit = 69.45;
+
+  // constants that define the maximum number of events in one transaction
+  const maxSequenceEventElements = 16;
+  const maxEventElements = 16;
+
+  let S = 'Signature';
+  let N = 'None_given';
+  let P = 'Proof';
+
+  const isPair = (pair: string) =>
+    pair == S + N || pair == N + S || pair == S + S || pair == N + N;
+
+  function filterPairs(xs: string[]): {
+    xs: string[];
+    pairs: number;
+  } {
+    if (xs.length <= 1)
+      return {
+        xs,
+        pairs: 0,
+      };
+    if (isPair(xs[0].concat(xs[1]))) {
+      let rec = filterPairs(xs.slice(2));
+      return {
+        xs: rec.xs,
+        pairs: rec.pairs + 1,
+      };
+    } else {
+      let rec = filterPairs(xs.slice(1));
+      return {
+        xs: [xs[0]].concat(rec.xs),
+        pairs: rec.pairs,
+      };
+    }
+  }
+
+  function filterGroups(xs: string[]): {
```

Comment:
it would be nice to add a couple of test cases for this to one of the unit tests!